test/sd-test-srv/main.go

```diff
@@ -0,0 +1,80 @@
+// sd-test-srv runs a simple service discovery system; it returns two hardcoded
+// IP addresses for every A query.
+package main
+
+import (
+	"flag"
+	"log"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/miekg/dns"
+)
+
+func dnsHandler(w dns.ResponseWriter, r *dns.Msg) {
+	m := new(dns.Msg)
+	m.SetReply(r)
+	m.Compress = false
+
+	if len(r.Question) < 1 {
+		m.Rcode = dns.RcodeServerFailure
+		w.WriteMsg(m)
+		return
+	}
+	if !strings.HasSuffix(r.Question[0].Name, ".boulder.") {
+		m.Rcode = dns.RcodeServerFailure
+		w.WriteMsg(m)
+		return
+	}
+
+	hdr := dns.RR_Header{
+		Name:   r.Question[0].Name,
+		Rrtype: dns.TypeA,
+		Class:  dns.ClassINET,
+		Ttl:    0,
+	}
+
+	m.Answer = append(m.Answer, &dns.A{
```

Comment:
Can you add a comment explaining the provenance of these two IPs? they correspond to the red/blue net correct?